bundle/Controller/Admin/AdminController.php

```diff
@@ -271,10 +289,81 @@ public function handleCollectionAction(Request $request): RedirectResponse
         return $this->redirectToRoute('netgen_information_collection.route.admin.view', ['collectionId' => $collectionId]);
     }
 
+    public function editAction(Request $request, int $collectionId)
+    {
+        $this->checkEditPermissions();
+
+        $collection = $this->service->getCollection(new CollectionId($collectionId));
+
+        $locationService = $this->getRepository()->getLocationService();
+
+        $location = $locationService->loadLocation($collection->getContent()->contentInfo->mainLocationId);
```

Comment:
Main Location is directly available on the `ContentInfo` object.

Reply:
fixed in 251ce386